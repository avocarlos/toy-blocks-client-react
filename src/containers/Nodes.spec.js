import React from "react";
import { shallow } from "enzyme";
import configureMockStore from "redux-mock-store";
import thunk from 'redux-thunk';
import { Provider } from "react-redux";
import { render } from "@testing-library/react";
import ConnectedNodes, { Nodes } from "./Nodes";
import Node from "../components/Node";

describe("<Nodes />", () => {
  const actions = {
    checkNodeStatuses: jest.fn()
  };

  const nodes = {
    list: [
      {
        url: 'https://thawing-springs-53971.herokuapp.com',
        online: false,
        name: 'Node 1',
        loading: false,
        blocks: {
          loading: false,
          error: false,
          list: [{    
            id: "5",
            type: "blocks",
            attributes: {
              index: 1,
              timestamp: 1530679678,
              data: "The Human Torch",
              "previous-hash": "KsmmdGrKVDr43/OYlM/oFzr7oh6wHG+uM9UpRyIoVe8=",
              hash: "oHkxOJWOKy02vA9r4iRHVqTgqT+Afc6OYFcNYzyhGEc="
            }
          }]
        }
      },
      {
        url: 'https://secret-lowlands-62331.herokuapp.com',
        online: false,
        name: 'Node 2',
        loading: false,
        blocks: {
          loading: false,
          error: false,
          list: []
        }
      }
    ]
  };

  it("should contain <Node />", () => {
    const wrapper = shallow(
      <Nodes
        actions={actions}
        nodes={nodes}
      />
    );

    expect(wrapper.find(Node).length).toEqual(2);
  });

  it("should match snapshot", () => {
    const middlewares = [thunk];
    const store = configureMockStore(middlewares)({nodes});
    const { asFragment } = render(
      <Provider store={store}>
        <ConnectedNodes />
      </Provider>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
