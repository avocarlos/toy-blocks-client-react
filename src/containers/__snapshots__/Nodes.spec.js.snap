// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Nodes /> should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root MuiBox-root-1"
  >
    <h1
      class="MuiTypography-root MuiTypography-h4"
    >
      <strong
        style="color: rgb(0, 0, 0);"
      >
        Nodes
      </strong>
    </h1>
    <div
      class="MuiPaper-root MuiAccordion-root makeStyles-root-2 MuiAccordion-rounded MuiPaper-elevation3 MuiPaper-rounded"
    >
      <div
        aria-disabled="false"
        aria-expanded="false"
        class="MuiButtonBase-root MuiAccordionSummary-root makeStyles-summary-3"
        role="button"
        tabindex="0"
      >
        <div
          class="MuiAccordionSummary-content makeStyles-content-6"
        >
          <div
            class="MuiBox-root MuiBox-root-11 makeStyles-summaryContent-4"
          >
            <div
              class="MuiBox-root MuiBox-root-12"
            >
              <h5
                class="MuiTypography-root makeStyles-heading-8 MuiTypography-h5"
              >
                Node 1
              </h5>
              <h6
                class="MuiTypography-root makeStyles-secondaryHeading-9 MuiTypography-subtitle1"
              >
                https://thawing-springs-53971.herokuapp.com
              </h6>
            </div>
            <div
              class="MuiBox-root MuiBox-root-17"
            >
              <span
                class="makeStyles-dot-13 makeStyles-dot-15"
              />
              <span
                class="makeStyles-text-14 makeStyles-text-16"
              >
                OFFLINE
              </span>
            </div>
          </div>
        </div>
        <div
          aria-disabled="false"
          aria-hidden="true"
          class="MuiButtonBase-root MuiIconButton-root MuiAccordionSummary-expandIcon makeStyles-icon-5 MuiIconButton-edgeEnd"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </div>
      </div>
      <div
        class="MuiCollapse-container MuiCollapse-hidden"
        style="min-height: 0px;"
      >
        <div
          class="MuiCollapse-wrapper"
        >
          <div
            class="MuiCollapse-wrapperInner"
          >
            <div
              role="region"
            >
              <div
                class="MuiAccordionDetails-root makeStyles-details-10"
              >
                <div
                  class="MuiBox-root MuiBox-root-19"
                >
                  <span
                    class="MuiTypography-root makeStyles-index-18 MuiTypography-caption"
                  >
                    001
                  </span>
                  <p
                    class="MuiTypography-root MuiTypography-body2"
                  >
                    The Human Torch
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiPaper-root MuiAccordion-root makeStyles-root-2 MuiAccordion-rounded MuiPaper-elevation3 MuiPaper-rounded"
    >
      <div
        aria-disabled="false"
        aria-expanded="false"
        class="MuiButtonBase-root MuiAccordionSummary-root makeStyles-summary-3"
        role="button"
        tabindex="0"
      >
        <div
          class="MuiAccordionSummary-content makeStyles-content-6"
        >
          <div
            class="MuiBox-root MuiBox-root-20 makeStyles-summaryContent-4"
          >
            <div
              class="MuiBox-root MuiBox-root-21"
            >
              <h5
                class="MuiTypography-root makeStyles-heading-8 MuiTypography-h5"
              >
                Node 2
              </h5>
              <h6
                class="MuiTypography-root makeStyles-secondaryHeading-9 MuiTypography-subtitle1"
              >
                https://secret-lowlands-62331.herokuapp.com
              </h6>
            </div>
            <div
              class="MuiBox-root MuiBox-root-24"
            >
              <span
                class="makeStyles-dot-13 makeStyles-dot-22"
              />
              <span
                class="makeStyles-text-14 makeStyles-text-23"
              >
                OFFLINE
              </span>
            </div>
          </div>
        </div>
        <div
          aria-disabled="false"
          aria-hidden="true"
          class="MuiButtonBase-root MuiIconButton-root MuiAccordionSummary-expandIcon makeStyles-icon-5 MuiIconButton-edgeEnd"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </div>
      </div>
      <div
        class="MuiCollapse-container MuiCollapse-hidden"
        style="min-height: 0px;"
      >
        <div
          class="MuiCollapse-wrapper"
        >
          <div
            class="MuiCollapse-wrapperInner"
          >
            <div
              role="region"
            >
              <div
                class="MuiAccordionDetails-root makeStyles-details-10"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
